### PR TITLE
Adding helpers to add/remove to/from Windows PATH

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**
I'm currently YOLO testing this <3

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit 8a7458226a0dd16af98cfc3e2c8a4a6227b335c7 (Job results: [836](https://jenkins.osquery.io/job/osqueryPullRequestWindows/836/)) failed one or more tests (**Windows**).

**Comment by theopolis** [comment]
Here an example of a fresh provision of 2012R2: https://jenkins.osquery.io/job/osqueryMasterBuildWindows2012R2/35/console it is failing because Python is not found.

File: tools/provision.ps1 [hunk 1]
```diff
@@ -112,13 +147,14 @@ function Install-Chocolatey {
...
-      Write-Host "[-] WARN: Chocolatey appears to be installed, but was not in path!" -foregroundcolor Yellow
+      Write-Host "[-] WARN: Chocolatey appears to be installed, but cannot be found in the system path!" -foregroundcolor Yellow
     } else {
       Write-Host " => Did not find. Installing chocolatey..." -foregroundcolor Cyan
       Invoke-Expression ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))
     }
     Write-Host " => Adding chocolatey to path."
-    $env:Path = "$env:Path;$env:ALLUSERSPROFILE\chocolatey\bin"
+    $chocoPath = "$env:ALLUSERSPROFILE\\chocolatey\\bin"
+    Append-Path $chocoPath
```
Comment (theopolis) [inline tools/provision.ps1]: `Add-ToPath`?

**Comment by osqueryer** [comment]
:-1: The commit 32f93eac3257f9c15ca808076694aab0772e647f (Job results: [843](https://jenkins.osquery.io/job/osqueryPullRequestWindows/843/)) failed one or more tests (**Windows**).

File: tools/provision.ps1 [hunk 2]
```diff
@@ -169,7 +205,10 @@ function Install-PipPackage {
     Exit -1
   }
   Write-Host " => Attempting to install Python packages" -foregroundcolor DarkYellow
-  $env:Path = "$env:Path;$env:HOMEDRIVE\tools\python2;$env:HOMEDRIVE\tools\python2\Scripts"
+  $pythonPath = "$env:HOMEDRIVE\\tools\\python2"
+  $pythonScriptsPath = "$env:HOMEDRIVE\\tools\\python2\\Scripts"
```
Comment (PoppySeedPlehzr) [inline tools/provision.ps1]: This isn't working. To match against strings that already exist in the path I need the `\\`, however when adding that character isn't treated as the escape character. Got something better coming here ina bit.

**Comment by osqueryer** [comment]
:-1: The commit faebb0d592abc12c496d0c800371cd2efe635c90 (Job results: [3848](https://jenkins.osquery.io/job/osqueryPullRequestLinux/3848/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit faebb0d592abc12c496d0c800371cd2efe635c90 (Job results: [845](https://jenkins.osquery.io/job/osqueryPullRequestWindows/845/)) failed one or more tests (**Windows**).

**Comment by theopolis** [comment]
Sorry, I may have messed this up with some Jenkins configurations.

test this please

**Comment by osqueryer** [comment]
:-1: The commit faebb0d592abc12c496d0c800371cd2efe635c90 (Job results: [846](https://jenkins.osquery.io/job/osqueryPullRequestWindows/846/)) failed one or more tests (**Windows**).

**Comment by PoppySeedPlehzr** [comment]
ok to test

**Comment by osqueryer** [comment]
:-1: The commit faebb0d592abc12c496d0c800371cd2efe635c90 (Job results: [847](https://jenkins.osquery.io/job/osqueryPullRequestWindows/847/)) failed one or more tests (**Windows**).

**Comment by theopolis** [comment]
I was also seeing:
```
 => Determining whether python2 is already installed...
[*] python2 2.7.11 already installed.
 => Attempting to install Python packages
[-] ERROR: failed to find pip
``` 
On Windows10 when the python paths weren't installed correctly.

**Comment by osqueryer** [comment]
:-1: The commit faebb0d592abc12c496d0c800371cd2efe635c90 (Job results: [3850](https://jenkins.osquery.io/job/osqueryPullRequestLinux/3850/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit faebb0d592abc12c496d0c800371cd2efe635c90 (Job results: [3851](https://jenkins.osquery.io/job/osqueryPullRequestLinux/3851/)) failed one or more tests (**Linux**).

**Comment by theopolis** [comment]
You can ignore the Linux build errors for now. I turned on additional testing for them and there are errors we need to fix, most likely in configuration/pack content.

**Comment by theopolis** [comment]
Feel free to ignore `Linux` errors, rebasing fixes them though.

**Pull request merged**